packages/wrangler/src/deploy/deploy.ts

```diff
@@ -168,6 +168,53 @@ function errIsStartupErr(err: unknown): err is ParseError & { code: 10021 } {
 	return false;
 }
 
+export const validateRoutes = (
+	routes: Route[],
+	hasExperimentalAssets: boolean
+) => {
+	const invalidRoutes: { [route: string]: string[] } = {};
+	for (const route of routes) {
+		if (typeof route !== "string" && route.custom_domain) {
+			if (route.pattern.includes("*")) {
+				invalidRoutes[route.pattern] ??= [];
+				invalidRoutes[route.pattern].push(
+					`Wildcard operators (*) are not allowed in Custom Domains`
+				);
+			}
+			if (route.pattern.includes("/")) {
+				invalidRoutes[route.pattern] ??= [];
+				invalidRoutes[route.pattern].push(
+					`Paths are not allowed in Custom Domains`
+				);
+			}
+		} else if (hasExperimentalAssets) {
+			const pattern = typeof route === "string" ? route : route.pattern;
+			const components = pattern.split("/");
+
+			if (
+				components.length === 1 ||
+				(components.length === 2 && components[1] === "")
+			) {
+				invalidRoutes[pattern] ??= [];
+				invalidRoutes[pattern].push(`Update the route to end with /*`);
```

Comment:
Having just the action stated here might be a bit too cryptic for users. Can we also tell them what routing rule they're actually breaking here? like we do in all other places.

Reply:
Ah I'm not sure what the phrasing should be - i stole this one from dash. tacking copy changes/discussion onto this https://jira.cfdata.org/browse/WC-2675

Reply:
Could we preserve the `Workers which have static assets cannot be routed on a URL which has a path component.` prefix here? The dash has a pattern of hiding the extra informational text when displaying errors (which is why it's so terse), but at the point that error is shown in dash that user should have been able to read the informational text.

Reply:
Ah yes, I just commented on this above. I agree it is not ideal copy.